### PR TITLE
Bug: tools: Fix moving a resource with a lifetime constraint

### PR DESCRIPTION
**Pull request by clumens**
A node can be specified two different ways - with attribute="#uname"
and value= on an expression XML node, or with node= on a rsc_location
XML node. Both possibilities need to be considered when attempting to
clear an existing constraint.

cli_resource_clear was previously only considering the second case.
This patch rearranges the code to allow for trying both, and then adds
the code to try the first case by constructing a different blob of XML
to match.

See rhbz#1648620

### CODE REVIEW

**Comment by clumens** [comment]
I'd still consider this a WIP.  I need more tests.  However, I wanted to get opinions on the general approach and patch #3 in particular.

**Comment by kgaillot** [comment]
I'm thinking this would be good to get into 2.0.1, so please resubmit against 2.0 branch

File: tools/crm_resource_ban.c [hunk 1]
```diff
@@ -191,51 +191,105 @@ cli_resource_prefer(const char *rsc_id, const char *host, cib_t * cib_conn)
...
 {
     int rc = pcmk_ok;
-    xmlNode *fragment = NULL;
-    xmlNode *location = NULL;
+    char *xpath_string = NULL;
 
-    if(cib_conn == NULL) {
-        return -ENOTCONN;
-    }
+    xpath_string = crm_strdup_printf("//rsc_location[@id='cli-prefer-%s'][rule[@id='cli-prefer-rule-%s']/expression[@attribute='#uname' and @value='%s']]",
```
[the remark this answers is not in the document]
Comment (clumens) [reply tools/crm_resource_ban.c]: If you look carefully, you'll see that the bracket starting at rule goes all the way to the end of the expression.  This syntax says you want to return the rsc_location node that has a child that is a rule with id= and its child is an expression with those attributes.  If it were just straight slashes all the way through, I'd be asking for the expression node.

**Comment by kgaillot** [comment]
I'll merge the 2.0 version of this, pull 2.0 into master after rc3 is released, and add a commit to master updating the test output at that time

**Pull request closed without merging**